pkg/controller/pattern2/service.go

```diff
@@ -0,0 +1,221 @@
+
+	return &corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "wso2-am-2-svc",
+			Namespace: apimanager.Namespace,
+			OwnerReferences: []metav1.OwnerReference{
+				*metav1.NewControllerRef(apimanager, apimv1alpha1.SchemeGroupVersion.WithKind("APIManager")),
+			},
+		},
+		Spec: corev1.ServiceSpec{
+			Selector: labels,
+			Ports:    pubdevtm2ports,
+		},
+	}
+}
+
+//GatewayService is for handling gateway-sevice...
+func GatewayService(apimanager *apimv1alpha1.APIManager) *corev1.Service {
+	labels := map[string]string{
+		"deployment": "wso2-gateway",
+	}
+	gatewayports := getGatewaySpecificSvcPorts()
+
+	return &corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "wso2-am-gw-svc",
+			Namespace: apimanager.Namespace,
+			OwnerReferences: []metav1.OwnerReference{
+				*metav1.NewControllerRef(apimanager, apimv1alpha1.SchemeGroupVersion.WithKind("APIManager")),
+			},
+		},
+		Spec: corev1.ServiceSpec{
+			Selector: labels,
+			Ports:    gatewayports,
+		},
+	}
+}
+
+//KeyManagerService is for handling key manager service...
+func KeyManagerService(apimanager *apimv1alpha1.APIManager) *corev1.Service {
+	labels := map[string]string{
+		"deployment": "wso2-km",
+	}
+	keymanagerports := getKeyManagerSpecificSvcPorts()
+
+	return &corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "wso2-am-km-svc",
+			Namespace: apimanager.Namespace,
+			OwnerReferences: []metav1.OwnerReference{
+				*metav1.NewControllerRef(apimanager, apimv1alpha1.SchemeGroupVersion.WithKind("APIManager")),
+			},
+		},
+		Spec: corev1.ServiceSpec{
+			Selector: labels,
+			Ports:    keymanagerports,
+		},
+	}
+}
+
+// DashboardService for handling analytics-dashboard service...
+func DashboardService(apimanager *apimv1alpha1.APIManager) *corev1.Service {
+	labels := map[string]string{
+		"deployment": "wso2-analytics-dashboard",
+	}
+	dashports := getDashBoardPorts()
+
+	return &corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "wso2-am-analytics-dashboard-svc",
+			Namespace: apimanager.Namespace,
+			OwnerReferences: []metav1.OwnerReference{
+				*metav1.NewControllerRef(apimanager, apimv1alpha1.SchemeGroupVersion.WithKind("APIManager")),
+			},
+		},
+		Spec: corev1.ServiceSpec{
+			Selector: labels,
+			Ports:    dashports,
+		},
+	}
+}
+
+//WorkerService is for handling analytics-worker service
+func WorkerService(apimanager *apimv1alpha1.APIManager) *corev1.Service {
+	labels := map[string]string{
+		"deployment": "wso2-analytics-worker",
+	}
+	workerports := getWorkerPorts()
+
+	return &corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "wso2-am-analytics-worker-svc",
+			Namespace: apimanager.Namespace,
+			OwnerReferences: []metav1.OwnerReference{
+				*metav1.NewControllerRef(apimanager, apimv1alpha1.SchemeGroupVersion.WithKind("APIManager")),
+			},
+		},
+		Spec: corev1.ServiceSpec{
+			Selector: labels,
+			Ports:    workerports,
+		},
+	}
+}
+
+//WorkerHeadlessService is for...
```

Comment:
Seems this comment is incomplete?

Reply:
addressed via https://github.com/wso2/k8s-wso2am-operator/pull/89/commits/788038d4b86839a0ede2e666405f0c4f501c0dbf